File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/InputsAndResultsTestCharacter.js
print("getCharacter")
if(typeof getCharacter !== 'undefined')
	throw "getCharacter already defined!"
getCharacter = putCharacter
getCharacterType = typeof getCharacter
print(getCharacterType)
print(getCharacter)
if(getCharacterType !== 'string')
	throw "getCharacter not string!"

print("getCharacterVector")
if(typeof getCharacterVector !== 'undefined')
	throw "getCharacterVector already defined!"
getCharacterVector = putCharacterVector
getCharacterVectorType = typeof getCharacterVector[0]
print(getCharacterVectorType)
print(getCharacterVector)
if(getCharacterVectorType !== 'string')
	throw "getCharacterVector not string!"

print("getCharacterVectorAsList")
if(typeof getCharacterVectorAsList !== 'undefined')
	throw "getCharacterVectorAsList already defined!"
getCharacterVectorAsList = putCharacterVectorAsList
getCharacterVectorAsListType = typeof getCharacterVectorAsList[0]
print(getCharacterVectorAsListType)
print(getCharacterVectorAsList)
if(getCharacterVectorAsListType !== 'string')
	throw "getCharacterVectorAsList not string!"

print("getCharacterMatrix")
if(typeof getCharacterMatrix !== 'undefined')
	throw "getCharacterMatrix already defined!"
getCharacterMatrix = putCharacterMatrix
getCharacterMatrixType = typeof getCharacterMatrix[0][0]
print(getCharacterMatrixType)
print(getCharacterMatrix)
if(getCharacterMatrixType !== 'string')
	throw "getCharacterMatrix not string!"

print("getCharacterMatrixAsList")
if(typeof getCharacterMatrixAsList !== 'undefined')
	throw "getCharacterMatrixAsList already defined!"
getCharacterMatrixAsList = putCharacterMatrixAsList
getCharacterMatrixAsListType = typeof getCharacterMatrixAsList[0][0]
print(getCharacterMatrixAsListType)
print(getCharacterMatrixAsList)
if(getCharacterMatrixAsListType !== 'string')
	throw "getCharacterMatrixAsList not string!"

File: invesdwin-context-parent/invesdwin-context-scala/src/test/java/de/invesdwin/context/scala/tests/InputsAndResultsTestCharacter.scala
System.out.println("getCharacter")
if(getCharacter != null)
	throw new Exception("getCharacter already defined!")
val getCharacter = putCharacter
val getCharacterType = getCharacter.getClass()
System.out.println(getCharacterType)
System.out.println(getCharacter)
if(getCharacterType != classOf[java.lang.Character])
	throw new Exception("getCharacter not Character!")

System.out.println("getCharacterVector")
if(getCharacterVector != null)
	throw new Exception("getCharacterVector already defined!")
val getCharacterVector = putCharacterVector.asInstanceOf[Array[Char]]
val getCharacterVectorType = getCharacterVector(0).getClass()
System.out.println(getCharacterVectorType)
System.out.println(getCharacterVector)
if(getCharacterVectorType != classOf[Char])
	throw new Exception("getCharacterVector not Character!")

System.out.println("getCharacterVectorAsList")
if(getCharacterVectorAsList != null)
	throw new Exception("getCharacterVectorAsList already defined!")
val getCharacterVectorAsList = putCharacterVectorAsList.asInstanceOf[Array[Char]]
val getCharacterVectorAsListType = getCharacterVectorAsList(0).getClass()
System.out.println(getCharacterVectorAsListType)
System.out.println(getCharacterVectorAsList)
if(getCharacterVectorAsListType != classOf[Char])
	throw new Exception("getCharacterVectorAsList not Character!")

System.out.println("getCharacterMatrix")
if(getCharacterMatrix != null)
	throw new Exception("getCharacterMatrix already defined!")
val getCharacterMatrix = putCharacterMatrix.asInstanceOf[Array[Array[Char]]]
val getCharacterMatrixType = getCharacterMatrix(0)(0).getClass()
System.out.println(getCharacterMatrixType)
System.out.println(getCharacterMatrix)
if(getCharacterMatrixType != classOf[Char])
	throw new Exception("getCharacterMatrix not Character!")

System.out.println("getCharacterMatrixAsList")
if(getCharacterMatrixAsList != null)
	throw new Exception("getCharacterMatrixAsList already defined!")
val getCharacterMatrixAsList = putCharacterMatrixAsList.asInstanceOf[Array[Array[Char]]]
val getCharacterMatrixAsListType = getCharacterMatrixAsList(0)(0).getClass()
System.out.println(getCharacterMatrixAsListType)
System.out.println(getCharacterMatrixAsList)
if(getCharacterMatrixAsListType != classOf[Char])
	throw new Exception("getCharacterMatrixAsList not Character!")